Assets/Scripts/CombatScripts/BulletShellController.cs

```diff
@@ -20,21 +20,34 @@ void Start()
     private void Update(){
 
         if(firePos != null && (Vector3.Distance(gameObject.transform.position, firePos) > 17f))
-            Destroy(bulletPrefab);
+            Destroy(gameObject);
     }
 
     private void Awake(){
         firePos = gameObject.transform.position;
         
         Rigidbody2D bulletRB = gameObject.GetComponent<Rigidbody2D>();
-        bulletRB.velocity = new Vector2(15f, 0);
+        GameObject player = GameObject.Find("Player");

```

Comment:
This is fine for now, but I would rather have us find the player using `[SerializeField]`

Reply:
I agree this is not ideal, but i also don't think we should reference the player here at all. I only made these quick and dirty changes to showcase a fix for an issue where we were using 2 separate bullet prefabs for different direction bullets. i think this whole function can actually be deleted and the velocity of the bullet should be set by the thing instantiating the bullet. I will do that tonight.

Reply:
Agreed.

Reply:
For now we can keep it like this, but I am going to add a comment on this line that this will need to change down the line. I am not a fan of looking for objects using Find(), I am sure there is a better way to do it, but we can test some things out later.